### PR TITLE
[WIP] Transfer NFT feature

### PR DESCRIPTION
**Pull request by naftalimurgor**
Hello @cryptnotiq   
Active work in progress for this https://github.com/FraktalNFT/dapp-frontend/issues/47
- Allow a user to transfer an NFT to a valid address feature
- Design and flow same as the Opensea transfer feature

### CODE REVIEW

**Comment by cryptnotiq** [comment]
@naftalimurgor awesome, please ping me when ready to review and merge

**Comment by naftalimurgor** [comment]
Hey @cryptnotiq 
Great! I will tag you immediately it's ready

File: components/nft-item/index.tsx [hunk 1]
```diff
@@ -81,15 +93,15 @@ const NFTItem = forwardRef<HTMLDivElement, NFTItemProps>(
...
+      if (BIamount.lt(utils.parseEther("1.0"))) {
+        showAmount = "<0.01";
+      } else {
+        showAmount = utils.formatEther(BIamount.div(100));
       }
-      
-    }else{
+
+    } else {
       showAmount = "";
```
Comment (GMSteuart) [inline components/nft-item/index.tsx:105]: Looks like this `else` block  is removable

File: components/nft-item/index.tsx [hunk 2]
```diff
@@ -38,41 +46,45 @@ interface NFTItemProps extends StackProps {
...
-    useEffect(()=>{
-      if(item){
-        getListingAmount(account,item.id,provider,marketAddress).then(
+    useEffect(() => {
+      if (item) {
+        getListingAmount(account, item.id, provider, marketAddress).then(
           (amount: BigNumber) => {
-            if(amount.gt(0)){
+            if (amount.gt(0)) {
               setIsListed(true);
```
Comment (GMSteuart) [inline components/nft-item/index.tsx:71]: Can remove the conditional here and use the following:
```
setIsListed(amount.gt(0))
```

File: pages/import-nft.tsx [hunk 3]
```diff
@@ -228,7 +228,7 @@ import { Workflow } from "types/workflow";
      ).then(receipt => {
```
Comment (GMSteuart) [inline pages/import-nft.tsx:228]: Can pull the success callback lines out into a separate function (e.g. `itemListedHandler`) that both `listNewItem` and `listNewAuctionItem` call to reduce the need for updating handler code in two spots.

Both functions would also benefit by being refactored into full use of async/await code structuring with try/catch blocks.

File: components/nft-item/index.tsx (hunk 1 above)
Comment (naftalimurgor) [reply components/nft-item/index.tsx:105]: Great insight. I'm on it.

File: pages/import-nft.tsx (hunk 3 above)
Comment (naftalimurgor) [reply pages/import-nft.tsx:228]: Great. I'm on it will push the change

Comment (naftalimurgor) [reply pages/import-nft.tsx:228]: Will resolve both locally then commit and push the changes shortly

**Comment by cryptnotiq** [comment]
hey @naftalimurgor great to see this coming along! Could you just make the transfer button go to the transfer page instead of triggering a dropdown where you have to click transfer a second time?

Let me know once it's ready to test out fully!

**Comment by naftalimurgor** [comment]
hello @cryptnotiq 
Great insight, thanks. Will do the changes to have the button get straight to the transfer page.

**Comment by naftalimurgor** [comment]
@cryptnotiq I will definitely let you know when it is fully ready. I'm thinking of having the "Transfer" button directly above the "FrakeIt" button with the same button style as the "FrakIt" button to make it look nicer and more consistent.

**Comment by cryptnotiq** [comment]
> @cryptnotiq I will definitely let you know when it is fully ready. I'm thinking of having the "Transfer" button directly above the "FrakeIt" button with the same button style as the "FrakIt" button to make it look nicer and more consistent.

I would keep it exactly how you have it now. Bc transfer is a secondary function that won't be used that often compared to the core button ones like Frak It. Looks perfect how it is imo

**Comment by naftalimurgor** [comment]
> > @cryptnotiq I will definitely let you know when it is fully ready. I'm thinking of having the "Transfer" button directly above the "FrakeIt" button with the same button style as the "FrakIt" button to make it look nicer and more consistent.
> 
> I would keep it exactly how you have it now. Bc transfer is a secondary function that won't be used that often compared to the core button ones like Frak It. Looks perfect how it is imo

I understand. It's not feature to be used quite often, It's good left the way it is

**Comment by cryptnotiq** [comment]
hey @naftalimurgor i'm going to close this PR for now to keep our repo organised, but please open it once again, once it is ready to be fully tested!

**Pull request closed without merging**